### PR TITLE
refactor(wallet-ci): Use `action-gh-release` action

### LINKED ISSUE #4268: Use `action-gh-release` in wallet prod build workflow
https://github.com/softprops/action-gh-release

### PR DESCRIPTION
**Pull request by marc2332**
Closes https://github.com/iotaledger/iota/issues/4268

### CODE REVIEW

**Review by msarcev — APPROVED** [review]

File: .github/workflows/apps_wallet_prod_build.yml [hunk 1]
```diff
@@ -59,27 +59,28 @@ jobs:
...
+            echo "changelog=$changelog" >> $GITHUB_OUTPUT
           fi
 
       - name: Get version from tag
         id: version
         run: echo "::set-output name=version::${GITHUB_REF#refs/tags/wallet-v}"
 
       - name: Create GitHub Release
-        uses: elgohr/Github-Release-Action@c5ea99036abb741a89f8bf1f2cd7fba845e3313a # pin@v5
+        uses: softprops/action-gh-release@01570a1f39cb168c169c802c3bceb9e93fb10974 # pin@v2.1.0
```
Comment (thibault-martinez) [inline .github/workflows/apps_wallet_prod_build.yml]: can you ELI5 the differences pls?

Comment (marc2332) [reply .github/workflows/apps_wallet_prod_build.yml]: Its just that for some reason I used https://github.com/elgohr/Github-Release-Action while thinking it was https://github.com/actions/create-release, and turns out elgohr's action barely has any supported params, that's why I am making the switch to softprops, which has way more params just like the old `actions/create-release`

Comment (thibault-martinez) [reply .github/workflows/apps_wallet_prod_build.yml]: Thank you kind sir

**Comment by marc2332** [comment]
example: https://github.com/iotaledger/iota/releases/tag/untagged-1ca3f866a69fd0efdaad

**Comment by DaughterOfMars** [comment]
What's the reason for the change?

File: .github/workflows/apps_wallet_prod_build.yml [hunk 2]
```diff
@@ -59,29 +59,27 @@ jobs:
...
+            git log ${{ env.PREV_TAG }}..${{ env.CURRENT_TAG }} --pretty=format:"- %s in #%h" -- ./apps/wallet > CHANGELOG.md
           fi
 
       - name: Get version from tag
         id: version
         run: echo "::set-output name=version::${GITHUB_REF#refs/tags/wallet-v}"
 
       - name: Create GitHub Release
-        uses: elgohr/Github-Release-Action@c5ea99036abb741a89f8bf1f2cd7fba845e3313a # pin@v5
+        uses: softprops/action-gh-release@01570a1f39cb168c169c802c3bceb9e93fb10974 # pin@v2.1.0
```
Comment (thibault-martinez) [inline .github/workflows/apps_wallet_prod_build.yml]:
```suggestion
        uses: softprops/action-gh-release@01570a1f39cb168c169c802c3bceb9e93fb10974 # v2.1.0
```

**Review by begonaalvarezd — APPROVED** [review]
